Cargo.toml

```diff
@@ -14,6 +14,7 @@ bytes = "0.5"
 flume = { version = "0.7", default-features = false }
 futures = { version = "0.3", optional = true }
 once_cell = "1.3"
+rand = "0.7.3"
```

Comment:
I was thinking it would need an incremental search per IP, but there's no reason it couldn't just be one  monotonically increasing number that wraps. I'll put it back to that.